### ISSUE TITLE
Individual functions in specific files should override functions in generic files

### ISSUE THREAD

**Issue by JalonSolov** [vlang/v#8278]
Yes, the title is... awkward, perhaps.  Hard to put the concept into a short line.

Basically, V currently supports `.v` files for V code, `.c.v` files for mixed C and V, `_<platform>.c.v` files for something specific to a particular platform (like Linux, Windows, FreeBSD, etc.).

The problem is that the functions in these files must all be named differently, or V will give you an error about duplicate function names.

I have suggested an alternative:  That functions from more specific files override the functions in more general files.  For example:

`foo.v` has these functions:
```v
fn a(a int) int
fn b(b int) int
fn c(c int) int
...
fn z(z int) int
```
In a plain `.v` file, we should only see 'pure' V code, instead of a mix of V code with `C.` calls, etc.  This would be the more generic, "high-level" version of the code.

Suppose someone decides they can make a version of `fn b` that is twice as fast with the C backend, if they just add the `C.` calls.  That would mean that `foo.c.v` only needs a single
```v
fn b(b int) int
```
defined, while the other 25 `fn`s are still defined in `foo.v`.  It also means that, for example, the JavaScript, x64, etc. routines are not affected by the C compiler specific code,

Even more specific, you find out that `fn b` in `foo.c.v` can be made even better on Linux with a small change, but that change won't work anywhere else.  That's fine, define the Linux specific `fn b` in a `foo_linux.c.v` file, and it overrides both of the others, but ONLY if you're using the C backend on Linux.

Implementation of this change should be a relatively simple change in the compiler.  It would just mean sorting the function definitions from most specific to least, and ignoring duplicates when they're defined at multiple levels, instead of throwing a 'duplicate function' error.

Duplicates at the same level will still be an error case.

As things stand now, if you wanted `fn b` set up as described above, you would wind up with 78 functions defined in 3 files, all with different names, instead of the 28 functions with only 26 names if this proposal is implemented.

**Comment by Larpon** [comment]
I can see more positive sides than negative with this approach at first sight.

One major downside is how it can affect debugging a function that has the same name in all supported targets... which one is the culprit? What function is actually used on the bug reporters machine? If the tooling fail to tell us which one it is it can get really hairy.

**Comment by JalonSolov** [comment]
Agreed, it could become... interesting.

Maybe some compiler magic would be useful here, such as `fn a` becomes `fn c_a` or `fn linux_c_a` by the time it gets to the backend?

Not sure how to solve it, but that could be once approach.

**Comment by spytheman** [comment]
When compiling with -g, the backtraces will have .v file paths and line numbers. It would not be a problem with actual debuggers, but can become confusing, when just reading code without help from something like vls.

**Comment by spytheman** [comment]
The overriding mechanism is implemented in ec196cf .
